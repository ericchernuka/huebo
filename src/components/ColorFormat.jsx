import React from 'react'
import PropTypes from 'prop-types'
import Label from './Label'

const ColorFormat = ({ children, copied, label, ...props }) => (
  <div className="color-format-block" {...props}>
    <Label>{copied ? `${label} copied to clipboard` : label}</Label>
    {children}
  </div>
)

ColorFormat.propTypes = {
  copied: PropTypes.bool,
  label: PropTypes.string.isRequired,
}

ColorFormat.defaultProps = {
  copied: false,
}

export default ColorFormat
