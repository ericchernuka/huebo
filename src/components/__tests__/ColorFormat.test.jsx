import React from 'react'
import { render } from 'react-testing-library'
import ColorFormat from '../ColorFormat'

describe('when not copied', () => {
  test('renders without "copied to clipboard" text', () => {
    const children = '255, 255, 255'
    const { container } = render(
      <ColorFormat label="RGB">{children}</ColorFormat>,
    )

    expect(container.textContent).not.toMatch('copied to clipboard')
    expect(container).toMatchSnapshot()
  })
})

describe('when copied', () => {
  test('renders with "copied to clipboard" text', () => {
    const children = '255, 255, 255'
    const { container } = render(
      <ColorFormat label="RGB" copied>
        {children}
      </ColorFormat>,
    )

    expect(container.textContent).toMatch('copied to clipboard')
    expect(container).toMatchSnapshot()
  })
})
