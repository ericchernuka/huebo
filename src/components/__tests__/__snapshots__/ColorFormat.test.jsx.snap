// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when copied renders with "copied to clipboard" text 1`] = `
<div>
  <div
    class="color-format-block"
  >
    <h2>
      RGB copied to clipboard
    </h2>
    255, 255, 255
  </div>
</div>
`;

exports[`when not copied renders without "copied to clipboard" text 1`] = `
<div>
  <div
    class="color-format-block"
  >
    <h2>
      RGB
    </h2>
    255, 255, 255
  </div>
</div>
`;
